R/build.gradle

```diff
@@ -0,0 +1,107 @@
+plugins {
+    id 'com.bmuschko.docker-remote-api'
+    id 'io.deephaven.project.register'
+    id 'io.deephaven.deephaven-in-docker'
+}
+
+evaluationDependsOn(':cpp-client')
+
+configurations {
+    cpp {}
+}
+
+dependencies {
+  cpp project(':cpp-client')
+}
+
+def prefix = '/opt/deephaven'
+
+// start a grpc-api server
+String randomSuffix = UUID.randomUUID().toString();
+deephavenDocker {
+    envVars.set([
+            'START_OPTS':'-Xmx512m -DAuthHandlers=io.deephaven.auth.AnonymousAuthenticationHandler'
+    ])
+    containerName.set "dh-server-for-r-${randomSuffix}"
+    networkName.set "r-test-network-${randomSuffix}"
+}
+
+def buildRClient = Docker.registerDockerTask(project, 'rClient') {
+    // Only tested on x86-64, and we only build dependencies for x86-64
+    platform = 'linux/amd64'
+
+    copyIn {
+        from(layout.projectDirectory) {
+            include 'r-build.sh'
+            include 'r-tests.sh'
+            include 'rdeephaven/DESCRIPTION'
+            include 'rdeephaven/LICENSE'
+            include 'rdeephaven/NAMESPACE'
+            include 'rdeephaven/README.md'
+            include 'rdeephaven/inst/**'
+            include 'rdeephaven/man/**'
+            include 'rdeephaven/etc/**'
+            include 'rdeephaven/R/**'
+            include 'rdeephaven/src/*.cpp'
+            include 'rdeephaven/src/Makevars'
+        }
```

Comment:
if i'm reading this right, we're excluding the following:
 * build/
 * build.gradle
 * build.properties
 * rdeephaven/.gitignore
 * src/CMakeLists.txt
 * src/.gitignore

Any thought as to instead just copying everything (minus `build/`) to keep this a bit more brief, less confusing if a new file needs to be added later?

Possibly like this:
```patch
diff --git a/R/build.gradle b/R/build.gradle
index ee6b2c1b82..a66752b6a0 100644
--- a/R/build.gradle
+++ b/R/build.gradle
@@ -32,18 +32,7 @@ def buildRClient = Docker.registerDockerTask(project, 'rClient') {
 
     copyIn {
         from(layout.projectDirectory) {
-            include 'r-build.sh'
-            include 'r-tests.sh'
-            include 'rdeephaven/DESCRIPTION'
-            include 'rdeephaven/LICENSE'
-            include 'rdeephaven/NAMESPACE'
-            include 'rdeephaven/README.md'
-            include 'rdeephaven/inst/**'
-            include 'rdeephaven/man/**'
-            include 'rdeephaven/etc/**'
-            include 'rdeephaven/R/**'
-            include 'rdeephaven/src/*.cpp'
-            include 'rdeephaven/src/Makevars'
+            exclude 'build'
         }
     }
     dockerfile {
@@ -57,24 +46,14 @@ def buildRClient = Docker.registerDockerTask(project, 'rClient') {
                         ${prefix}/bin/rdeephaven \\
                         ${prefix}/src/rdeephaven/{inst,man,etc,src,R,bin}
                    """)
-        copyFile('rdeephaven/DESCRIPTION',  "${prefix}/src/rdeephaven/")
-        copyFile('rdeephaven/LICENSE',  "${prefix}/src/rdeephaven/")
-        copyFile('rdeephaven/NAMESPACE',  "${prefix}/src/rdeephaven/")
-        copyFile('rdeephaven/README.md',  "${prefix}/src/rdeephaven/")
-        copyFile('rdeephaven/inst/',  "${prefix}/src/rdeephaven/inst/")
-        copyFile('rdeephaven/man/', "${prefix}/src/rdeephaven/man/")
-        copyFile('rdeephaven/etc/', "${prefix}/src/rdeephaven/etc/")
-        copyFile('rdeephaven/R/', "${prefix}/src/rdeephaven/R/")
-        copyFile('rdeephaven/src/*.cpp', "${prefix}/src/rdeephaven/src/")
-        copyFile('rdeephaven/src/Makevars', "${prefix}/src/rdeephaven/src/")
+        copyFile('rdeephaven',  "${prefix}/src/rdeephaven")
         copyFile('r-tests.sh', "${prefix}/bin/rdeephaven")
         copyFile('r-build.sh', "${prefix}/bin/rdeephaven")
-        runCommand("PREFIX=\"" + prefix + "\"; \\\n" +
-                  '''set -eux ; \
-                     cd "${PREFIX}/src/rdeephaven"; \
-                     . "${PREFIX}/env.sh"; \
-                     ${PREFIX}/bin/rdeephaven/r-build.sh
-                   ''')
+        runCommand("""set -eux ; \
+                     cd "${prefix}/src/rdeephaven"; \
+                     . "${prefix}/env.sh"; \
+                     ${prefix}/bin/rdeephaven/r-build.sh
+                   """)
     }
     parentContainers = [ project.tasks.getByPath(':cpp-client:cppClient') ]
 }
```